sktime/forecasting/compose/_pipeline.py

```diff
@@ -114,6 +76,196 @@ def named_steps(self):
         """Map the steps to a dictionary"""
         return dict(self.steps)
 
+    def get_params(self, deep=True):
+        """Get parameters for this estimator.
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+        return self._get_params("steps", deep=deep)
+
+    def set_params(self, **kwargs):
+        """Set the parameters of this estimator.
+        Valid parameter keys can be listed with ``get_params()``.
+        Returns
+        -------
+        self
+        """
+        self._set_params("steps", **kwargs)
+        return self
+
+
+class ForecastingPipeline(_Pipeline):
+    """
+    Pipeline for forecasting with exogenous data to apply transformers
+    to the exogenous serieses. The forecaster can also be a
+    TransformedTargetForecaster containing transformers to
+    transform y. ForecastingPipeline is only applying the given transformers
+    to X.
+
+    Parameters
+    ----------
+    steps : list
+        List of tuples like ("name", forecaster/transformer)
+
+    Example
+    -------
+    >>> from sktime.datasets import load_longley
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.compose import ForecastingPipeline
+    >>> from sktime.transformations.series.impute import Imputer
+    >>> from sktime.transformations.series.adapt import TabularToSeriesAdaptor
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.model_selection import temporal_train_test_split
+    >>> from sklearn.preprocessing import MinMaxScaler
+    >>> y, X = load_longley()
+    >>> y_train, _, X_train, X_test = temporal_train_test_split(y, X)
+    >>> fh = ForecastingHorizon(X_test.index, is_relative=False)
+    >>> pipe = ForecastingPipeline(steps=[
+    ...     ("imputer", Imputer(method="mean")),
+    ...     ("minmaxscaler", TabularToSeriesAdaptor(MinMaxScaler())),
+    ...     ("forecaster", NaiveForecaster(strategy="drift"))])
+    >>> pipe.fit(y_train, X_train)
+    ForecastingPipeline(...)
+    >>> y_pred = pipe.predict(fh=fh, X=X_test)
+    """
+
+    _required_parameters = ["steps"]
+
+    def __init__(self, steps):
+        self.steps = steps
+        self.steps_ = None
+        super(ForecastingPipeline, self).__init__()
+
```

Comment:
@fkiraly and I found out that we need to run this in init too so that people can set nested parameters before calling fit.  
```suggestion
        self.steps_ = self._check_steps()
        
```